src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2109,6 +2111,45 @@ exprt c_typecheck_baset::do_special_functions(
 
     return same_object_expr;
   }
+  else if(identifier == CPROVER_PREFIX "points_to_valid_memory")
+  {
+    if(expr.arguments().size() != 2)
+    {
+      error().source_location = f_op.source_location();
+      error() << "points_to_valid_memory expects two operands" << eom;
+      throw 0;
+    }
+    if(!is_lvalue(expr.arguments().front()))
+    {
+      error().source_location = f_op.source_location();
+      error() << "ptr argument to points_to_valid_memory"
+              << " must be an lvalue" << eom;
+      throw 0;
+    }
+
+    exprt same_object_expr;
+    if(expr.arguments().size() == 2)
+    {
+      if(
+        expr.arguments()[1].type() != ID_unsignedbv &&
+        expr.arguments()[1].type() != ID_signedbv)
+      {
+        err_location(f_op);
+        error() << "size argument to points_to_valid_memory"
+                << " must be coercible to size_t" << eom;
+        throw 0;
+      }
+      same_object_expr =
+        points_to_valid_memory(expr.arguments()[0], expr.arguments()[1]);
+    }
+    else
+    {
+      UNREACHABLE;
+    }
+    same_object_expr.add_source_location() = source_location;
```

Comment:
Maybe `points_to_valid_memory` should actually take a `source_locationt` as an additional argument?